dkfans/keeperfx
# Add new Creature spell - Ranged Heal

The caster can use this spell to heal itself or other keeper's creature.
When not in combat, the caster will heal the wounded creature and
continue its original task/job.
When in combat, the caster will heal others only when not in melee
combat. After the casting, the caster will continue the combat.

Most parameters, like action time, reset time, of Heal and Ranged Heal
are the same.

This spell can be also used in the Possession mode.
To use it, click the number of this spell, move the mouse to put the
target in the center of the screen, and then click the left button.

Test Instructions:
1. To make the test easier, the following modifications are recommended.
  * Modify Druid to shorten its HungerRate.
  * Modify instance RANGED_HEAL to shorten its ResetTime.
You need a badly wound Orc (HP < 50%) to test.
2. Put a Lv9 Druid in the Training room. Put an Orc near the  Druid.
   Verify the Druid heals it and continue training.
3. Wait for Druid to go eating. On its way to Hatchery, drop an Orc,
   Verify the Druid heals it and then continues its way to Hatchery.
4. Wait for Druid to complete eating. Verify it goes back to the Training room.
5. Arrange a ranged combat for Druid, ensuring the enemy cannot reach
   the Druid. Drop an Orc nearby. Verify Druid heals Orc and resumes the
   fight.
6. Arrange a melee combat for an Orc and Knight. Drop a Druid nearby.
   Verify the Druid will heal the Orc and fight the Knight.
7. Arrange a melee combat for a Druid. Verify the Druid heals itself
   during combat.
8. Drop a wounded Orc near the Druid. Possess the Druid to control it.
   Verify you can use this spell to heal the Orc.
9. Try to hurt a Druid. Ensure the Ranged Heal becomes available during
   non-combat. Verify the Druid heals itself.
   You can just use the map editor to put a Druid with 20% HP, and then
   start the game. The Druid should heal itself soon.

Type: New Feature
